File: 8_Kyu_Kata/fake-binary/Python/fake_binary_test.py
# [object Object] - Fake Binary  [ ID: 57eae65a4321032ce000002d  (fake-binary) ]
# URL: https://www.codewars.com/kata/57eae65a4321032ce000002d
# Category: REFERENCE  |  Tags: FUNDAMENTALS | STRINGS | ARRAYS
# ******************************************************************************
# New tests for 3.8+
import codewars_test as test
from solution import fake_bin

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        tests = [
            # [expected, input]
            ["01011110001100111", "45385593107843568"],
            ["101000111101101", "509321967506747"],
            ["011011110000101010000011011", "366058562030849490134388085"],
            ["01111100", "15889923"],
            ["100111001111", "800857237867"],
        ]
        
        for exp, inp in tests:
            test.assert_equals(fake_bin(inp), exp)
    
@test.describe("Random Tests")
def random_tests():
    
    from random import randint
        
    def reference(x):
        return "".join('10'[int(c) <5] for c in x)
        
    for _ in range(100):
        test_case = "".join(str(randint(0, 9)) for _ in range(randint(1, 30)))
        @test.it(f"testing for fake_bin({test_case})")
        def test_case():
            test.assert_equals(fake_bin(test_case), reference(test_case))

File: kata-8-kyu/fake-binary/javascript/fakeBinary.Test.js
//+ ====================================================================================================================
//+
//+ 8 kyu - Fake Binary  [ ID: 57eae65a4321032ce000002d ] (fake-binary)
//+ URL: https://www.codewars.com/kata/57eae65a4321032ce000002d
//+ Category: REFERENCE  |  Tags: FUNDAMENTALS | STRINGS | ARRAYS
//+
//+ ====================================================================================================================

const { assert } = require("chai")
const { fakeBin } = require("./fakeBinary")

describe("Basic tests", () => {
  it("Testing for fixed tests", () => {
    assert.strictEqual(fakeBin("45385593107843568"), "01011110001100111")
    assert.strictEqual(fakeBin("509321967506747"), "101000111101101")
    assert.strictEqual(fakeBin("366058562030849490134388085"), "011011110000101010000011011")
    assert.strictEqual(fakeBin("15889923"), "01111100")
    assert.strictEqual(fakeBin("800857237867"), "100111001111")
  })
})

describe("Random tests", () => {
  const randint = (a, b) => ~~(Math.random() * (b - a + 1)) + a

  function myfakeBin(x) {
    x = x.split("")
    for (let i = 0; i < x.length; i++) {
      if (x[i] == "0" || x[i] == "1" || x[i] == "2" || x[i] == "3" || x[i] == "4") x[i] = "0"
      else x[i] = "1"
    }
    return x.join("")
  }

  let names = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"]

  it("Testing for 100 random tests", () => {
    for (let i = 0; i < 100; i++) {
      let x = [],
        len = randint(1, 30)
      for (let k = 0; k < len; k++) x.push(names[randint(0, names.length - 1)])
      x = x.join("")
      assert.strictEqual(fakeBin(x), myfakeBin(x), `Testing for ${JSON.stringify(x)}`)
    }
  })
})
